### ISSUE TITLE
CI emits torch inductor UserWarning: TensorFloat32 tensor cores for float32 matrix multiplication available but not enabled

### ISSUE THREAD

**Issue by albertvillanova** [huggingface/trl#6849]
### What

CI emits a new `UserWarning` from `torch._inductor` in the Liger tests:

```
tests/test_dpo_trainer.py::TestDPOTrainerVLM::test_train_vlm_liger
tests/test_kto_trainer.py::TestKTOTrainer::test_train_with_liger
tests/test_dpo_trainer.py::TestDPOTrainer::test_train_with_liger
tests/test_kto_trainer.py::TestKTOTrainerVLM::test_train_vlm_liger
  /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/_inductor/compile_fx.py:321: UserWarning: TensorFloat32 tensor cores for float32 matrix multiplication available but not enabled. Consider setting `torch.set_float32_matmul_precision('high')` for better performance.
    warnings.warn(
```

CI run: https://github.com/huggingface/trl/actions/runs/32414561958/job/96572632322

Affected tests:

- `tests/test_dpo_trainer.py::TestDPOTrainer::test_train_with_liger`
- `tests/test_dpo_trainer.py::TestDPOTrainerVLM::test_train_vlm_liger`
- `tests/test_kto_trainer.py::TestKTOTrainer::test_train_with_liger`
- `tests/test_kto_trainer.py::TestKTOTrainerVLM::test_train_vlm_liger`

The tests themselves pass; for now this is warning noise only.

### When it started

First seen in the run for #6741 (`1701d3e`), which moved the GPU CI from T4 to L40S: https://github.com/huggingface/trl/actions/runs/32366392208/job/96416712686 (4 warnings, all of them this one). No warning was reported in the run for `06c89977`, on the previous runner: https://github.com/huggingface/trl/actions/runs/32348545011/job/96362435182.

### Next steps

Investigate the root cause, then decide whether anything is actionable on the TRL side and/or should be reported upstream.

**Comment by albertvillanova** [comment]
## Root cause pinned down

### Why the runner change surfaced it

`torch._inductor` emits an advisory on every compile when TF32 tensor cores exist but the float32 matmul precision is left at its default — `torch/_inductor/compile_fx.py:315` ([v2.13.0](https://github.com/pytorch/pytorch/blob/v2.13.0/torch/_inductor/compile_fx.py#L315), the version CI installs; the `warnings.warn` is at line 321, matching the reported location):

```python
def _warn_tf32_disabled() -> None:
    if (
        torch.cuda.is_available()
        and torch.backends.cuda.matmul.fp32_precision != "tf32"
        and torch.cuda.get_device_capability() >= (8, 0)
    ):
        warnings.warn(
            "TensorFloat32 tensor cores for float32 matrix multiplication available but not enabled. "
            "Consider setting `torch.set_float32_matmul_precision('high')` for better performance."
        )
```

The `>= (8, 0)` guard is the whole story: T4 is sm75, so the advisory was unreachable on the old runner; L40S is sm89, so it now fires. Nothing else changed — the default precision has always been `"highest"`, and #6741 only swapped the hardware.

There is a second emission site with the same message in the fused-attention pattern matcher (`torch/_inductor/fx_passes/fuse_attention.py`), not reached by these tests.

### Why only the Liger tests

The advisory is reachable only from an inductor compile, and these four tests are the only ones in the suite that compile: Liger's chunked-loss base classes wrap the fused forward/backward in `torch.compile` when `compiled=True`, which is the default —

- `liger_kernel/chunked_loss/fused_linear_preference.py:193` (DPO)
- `liger_kernel/chunked_loss/fused_linear_unpaired_preference.py:157` (KTO)

The SFT Liger path uses the fused-linear-cross-entropy Triton kernels directly, with no `torch.compile`, which is why `test_train_with_liger` in the SFT tests is not affected.

### Nothing actionable on the TRL side

The advisory is addressed to the end user, and the thing it asks for — `torch.set_float32_matmul_precision('high')` — is a **process-wide numerics setting**. A library must not set it on its users' behalf: it would silently downgrade every float32 matmul in the process, including in code that has nothing to do with TRL. So this is not something to "fix" in `trl/`.

Setting it in `tests/conftest.py` would silence CI, but it would also change the numerics of every float32 matmul under our tolerance-based training tests, for the sole benefit of removing a log line. Not worth the risk.

Also note the message is purely informational: compilation and execution succeed either way, and it is emitted per compile with no dedup.

### Proposed solution

Upstream, both already filed by others, and neither is fixed:

- [pytorch#175484](https://github.com/pytorch/pytorch/issues/175484) (open) — asks for a way to opt out, and makes exactly the point above: *"I want my fp32 matrix multiplications to be in fp32, not tf32. But there is no avenue to suppressing this warning besides filtering logs."* `torch.set_float32_matmul_precision("highest")` cannot silence it, because torch does not distinguish the default `"highest"` from a user-set `"highest"`.
- [pytorch#185541](https://github.com/pytorch/pytorch/pull/185541) (closed, unmerged) — proposed demoting both advisories from `warnings.warn()` to `log.info()`, so they stay available to users collecting inductor logs without surfacing as Python warnings.

Nothing to add upstream that is not already said in #175484, so no new report from our side. What remains is deciding what we do in CI in the meantime.

### Next steps

- [ ] Decide what to do on the TRL side.

**Comment by albertvillanova** [comment]
Nothing is actionable on our side: the advisory asks for `torch.set_float32_matmul_precision('high')`, a process-wide numerics setting that a library must not change for its users, and there is no way to opt out (pytorch#175484). Meanwhile, I propose to silence it with a `filterwarnings` entry in `pyproject.toml`, documented and linked like the other upstream warning filters, so it can be removed once no supported `torch` version emits it.

**Comment by albertvillanova** [comment]
I've opened:
- #6851
